pydata_google_auth/auth.py

```diff
@@ -311,6 +312,19 @@ def get_user_credentials(
     pydata_google_auth.exceptions.PyDataCredentialsError
         If unable to get valid user credentials.
     """
+
+    # If the user isn't explicitly asking to authenticate with a particular
+    # client ID, then we should try to authenticate the user with Colab-based
+    # credentials, if possible.
+    if client_id is None:
```

Comment:
Should we also include the condition that this is not Colab Enterprise? I thought that was the original complaint that we are getting unnecessary warnings/errors there, where colab auth is known to not work.

Reply:
Never mind, in the Colab Enterprise ideally `get_user_credential` will not even be called (ADC will be returned at line 149)

Reply:
Correct